guides/source/active_record_callbacks.md

```diff
@@ -837,58 +837,59 @@ lead to invalid data.
 [`upsert_all`]:
     https://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-upsert_all
 
-Suppressing Callbacks
----------------------
+Suppressing Saving
+------------------
 
-In certain scenarios, you may need to temporarily prevent certain callbacks from
-being executed within your Rails application. This can be useful when you want
-to skip specific actions during certain operations without permanently disabling
-the callbacks.
+In certain scenarios, you may need to temporarily prevent records from being saved within your Rails application.
+This can be useful to skip the callbacks that would run if the record is saved successfully.
 
-Rails provides a mechanism for suppressing callbacks using the
-[`ActiveRecord::Suppressor`
-module](https://api.rubyonrails.org/classes/ActiveRecord/Suppressor.html). By
-using this module, you can wrap a block of code where you want to suppress
-callbacks, ensuring that they are not executed during that specific operation.
+Rails provides a mechanism to prevent saving records using the [`ActiveRecord::Suppressor` module]
+(https://api.rubyonrails.org/classes/ActiveRecord/Suppressor.html).
+By using this module, you can wrap a block of code where you want to avoid saving records and running their save callbacks.
 
-Let's consider a scenario where we have a `User` model with a callback that
-sends a welcome email to new users after they sign up. However, there might be
-cases where we want to create a user without sending the welcome email, such as
-during seeding the database with test data.
+Let's consider a scenario where we have a `User` `has_one` `Notification`.
+Creating a `User` will automatically create a `Notification` record as well.
+Saving a `Notification`, in turn, will trigger a callback that sends an email.
 
 ```ruby
 class User < ApplicationRecord
-  after_create :send_welcome_email
+  has_one :notification
+  after_create :create_notification
+end
+
+class Notification < ApplicationRecord
+  belongs_to :user
+
+  after_save :send_email
 
   def send_welcome_email
-    puts "Welcome email sent to #{self.email}"
+    puts "Sending welcome email"
   end
 end
 ```
 
-In this example, the `after_create` callback triggers the `send_welcome_email`
-method every time a new user is created.
+In this example, the `after_save` callback of `Notification` triggers the `send_email` method every time it is saved successfully.
 
-To create a user without sending the welcome email, we can use the
+To suppress the `after_save :send_email` callback, we can use the
 `ActiveRecord::Suppressor` module as follows:
```

Comment:
```suggestion
To prevent the `after_save :send_email` callback, we can suppress
saving of `Notification` records with `ActiveRecord::Suppressor`:
```